[2.3] magento catalog:images:resize now Database Media Storage mode aware

gwharton opened the pull request:
### Description (*)
Since the catalog:images:resize function was rewritten in 2.3, it is no longer database media storage mode aware. Resized images were neither extracted from the database if they didn't exist locally prior to resizing, nor were they stored back into the database once the resize was complete.

This PR fixes this.

### Fixed Issues (if relevant)
1. magento/magento2#23594 Database Media Storage : php bin/magento catalog:images:resize fails when image does not exist locally
2. magento/magento2#23595 Database Media Storage : php bin/magento catalog:images:resize fails to generate cached images in database
3. magento/magento2#23596 Database Media Storage : Add new product image, cached images not generated

### Manual testing scenarios (*)
1. deploy 2.3-develop
2. Change system storage mode to Database Media Storage mode, Synchronize and Save
3. Clear cache
4. Create test product, assign test product an image, and save product.
5. Check that product image appears in pub/media/catalog/product
6. Check that product cached images appear in pub/media/catalog/product/cache (Testing #23596)
7. Check that product image appears in database
8. Check that product cached images appear in database (Testing #23596)
9. Clear pub/media/catalog/product folder completely
10. Clear cached product images from database, leaving main product image intact
11. Run php bin/magento catalog:images:resize (Testing #23594)
12. Check that product image appears in pub/media/catalog/product
13. Check that product cached images appear in pub/media/catalog/product/cache
15. Check that product cached images appear in database (Testing #23595)

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

gwharton commented:
These problems are specific to 2.3 so no backporting is required.

ihor-sviziev commented:
@magento run tests

ihor-sviziev commented:
@magento run all tests

ihor-sviziev reviewed (dismissed):
Hi @gwharton,
From first view - changes looks good to me.
There were issues with running tests, they were re-ran. Waiting for results...

Would be great to cover your changes with tests. Could you do that?

Commit:
php bin/magento catalog:images:resize extracts source file from database
if in media storage mode and also stores resized images back into database
after resize.

magento-engcom-team commented:
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-5431](https://jira.corp.magento.com/browse/ENGCOM-5431) has been created to process this Pull Request

engcom-Alfa commented:
:heavy_check_mark: QA Passed
